man/LinReg.Rd

```diff
@@ -0,0 +1,90 @@
+\name{LinReg}
+\alias{LinReg}
+\alias{LinRegLA}
+\title{Simple Linear Regression}
+\usage{
+LinReg(model, particles = 1000, plot = FALSE)
+
+LinRegLA(model, particles = 1000, temperatures = seq(0, 1, 0.05)^5)
+}
+\arguments{
+\item{model}{Choice of regression model (1 for density as the predictor
+and 2 for adjusted density as the predictor).}
+
+\item{particles}{An integer specifying the number of particles.}
+
+\item{plot}{A boolean variable to determine whether to plot the posterior estimates.}
+
+\item{temperatures}{In likelihood annealing SMC the targets are defined as \eqn{P(y|\theta)^{\gamma_t}P(\theta)}
+where \eqn{0=\gamma_0\le \ldots \le \gamma_T = 1} can be referred to
+as the temperatures, \eqn{P(y|\theta)} is the likelihood and \eqn{P(\theta)}
+is the prior.}
+}
+\value{
+The \code{LinReg} function returns a \code{list} containing the final particle
+approximation to the target (\eqn{\theta} and the corresponding weights) as well as the logarithm
+of the estimated model evidence.
+
+The \code{LinRegLA} function returns a \code{list} containing the population
+of particles and their associates log likelihoods, log priors and weights at each iteration. The
+effective sample size at each of the iterations and several different
+estimates of the logarithm of the model evidence are also returned.
+}
+\description{
+A simple example based on estimating the parameters of a linear regression model
+using
+
+* Data annealing sequential Monte Carlo (\code{LinReg}).
+
+* Likelihood annealing sequential Monte Carlo (\code{LinRegLA}).
+}
+\details{
+Williams (1959) considers two competing linear regression models
+for the maximum compression strength parallel to the grain for radiata
+pine. Both models are of the form
+
+\eqn{y_i = \alpha + \beta (x_i - \bar{x}) + \epsilon_i},
+
+where \eqn{\epsilon_i ~ N(0,\sigma^2)} and \eqn{i=1,\ldots,42}.
+Here \eqn{y} is the maximum compression strength and \eqn{x} is either the density
+(model 1) or the adjusted density (model 2).
```

Comment:
My thinking on this is that most people reading it are likely to be familiar with the maths and so it shouldn't be too offputting. What we want is for someone looking at the code to be able to understand what it does without consulting external references except where it's essential.

Actually, the two equations are the same up to the labelling of the variables and the values upon which you are regression -- that is what you exploit in the code -- so my inclination would just be to tell people what the predictor is in a little bit more detail (just tell people what the density is the density of and what adjusted density means).